lib/Sema/TypeCheckAccess.cpp

```diff
@@ -1866,8 +1865,27 @@ class ExportabilityChecker : public DeclVisitor<ExportabilityChecker> {
                 getDiagnoseCallback(ED));
     });
 
-    if (!ED->getInherited().empty())
+    bool hasPublicMembers = llvm::any_of(ED->getMembers(),
+                                         [](const Decl *member) -> bool {
+      auto *valueMember = dyn_cast<ValueDecl>(member);
+      if (!valueMember)
+        return false;
+      return !shouldSkipChecking(valueMember);
+    });
+
+    if (hasPublicMembers) {
+      checkType(ED->getExtendedTypeLoc(), ED,
+                getDiagnoseCallback(ED), getDiagnoseCallback(ED));
+      // FIXME: include a note as to why this restriction is being enforced,
+      // i.e. there are public members.
```

Comment:
Are you planning to leave this as a FIXME until this turns into a public-facing feature?

Reply:
Yeah. :-/ It's what was there before this PR, and while it's not even that hard it's also not necessarily worth my time at the moment.